File: dotnet/src/xjustiz.core-dotnet/Models/Entities/Core/VersicherungCore.cs
namespace xjustiz.core_dotnet.Models.Entities.Core;

using System.Collections.Generic;
using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Helpers;

/// <summary>
/// Repräsentiert eine VersicherungCore.<br/>
/// <u><b>Insurance:</b></u> Represents an insurance.
/// </summary>
public class VersicherungCore
{
    /// <summary>
    /// Die Versicherungsnummer.<br/>
    /// <u><b>Insurance number:</b></u> The insurance number.
    /// </summary>
    [XmlElement("versicherungsnummer", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? Versicherungsnummer { get; set; }

    /// <summary>
    /// Liste der Versicherungsunternehmen.<br/>
    /// <u><b>Insurance companies:</b></u> List of insurance companies.
    /// </summary>
    [XmlElement("versicherungsunternehmen", Namespace = SchemeConstants.XJustizCore_Tns)]
    public List<Organisation>? Versicherungsunternehmen { get; set; }

    /// <summary>
    /// Die Anschrift der VersicherungCore.<br/>
    /// <u><b>Address:</b></u> The address of the insurance.
    /// </summary>
    [XmlElement("anschrift", Namespace = SchemeConstants.XJustizCore_Tns)]
    public Anschrift? Anschrift { get; set; }

    /// <summary>
    /// Die Referenz-ID.<br/>
    /// <u><b>Reference ID:</b></u> The reference ID.
    /// </summary>
    [XmlElement("referenzid", Namespace = SchemeConstants.XJustizCore_Tns)]
    public Guid? ReferenzId { get; set; }

    /// <summary>
    /// Der Versicherte.<br/>
    /// <u><b>Insured person:</b></u> The insured person.
    /// </summary>
    [XmlElement("versicherter", Namespace = SchemeConstants.XJustizCore_Tns)]
    public AuswahlBeteiligter? Versicherter { get; set; }

    /// <summary>
    /// Die Schadensnummer.<br/>
    /// <u><b>Claim number:</b></u> The claim number.
    /// </summary>
    [XmlElement("schadensnummer", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? Schadensnummer { get; set; }

    /// <summary>
    /// Der Typ der VersicherungCore.<br/>
    /// <u><b>Insurance type:</b></u> The type of insurance.
    /// </summary>
    [XmlElement("versicherungstyp", Namespace = SchemeConstants.XJustizCore_Tns)]
    public VersicherungsTyp? Versicherungstyp { get; set; }

    /// <summary>
    /// Der Typ der Rechtsschutzversicherung.<br/>
    /// <u><b>Legal protection type:</b></u> The type of legal protection insurance.
    /// </summary>
    [XmlElement("rechtsschutsversicherungstyp", Namespace = SchemeConstants.XJustizCore_Tns)]
    public RechtschutzVersicherungsTyp? RechtsschutzTyp { get; set; }

    /// <summary>
    /// Das versicherte Fahrzeug.<br/>
    /// <u><b>Vehicle:</b></u> The insured vehicle.
    /// </summary>
    [XmlElement("fahrzeug", Namespace = SchemeConstants.XJustizCore_Tns)]
    public Fahrzeug? Fahrzeug { get; set; }

    /// <summary>
    /// Informationen zur Selbstbeteiligung.<br/>
    /// <u><b>Deductible:</b></u> Information about the deductible.
    /// </summary>
    [XmlElement("selbstbeteiligung", Namespace = SchemeConstants.XJustizCore_Tns)]
    public Selbstbeteiligung? Selbstbeteiligung { get; set; }

    /// <summary>
    /// Anwendungsspezifische Erweiterungen.<br/>
    /// <u><b>Application-specific extensions:</b></u> Application-specific extensions.
    /// </summary>
    [XmlElement("anwendungsspezifischeErweiterung", Namespace = SchemeConstants.XJustiz_Tns)]
    public AnwendungsspezifischeErweiterung? AnwendungsspezifischeErweiterung { get; set; }
}

/// <summary>
/// Typ der VersicherungCore.<br/>
/// <u><b>Insurance type:</b></u> Insurance type.
/// </summary>
public class VersicherungsTyp
{
    /// <summary>
    /// Der Code.<br/>
    /// <u><b>Code:</b></u> The code.
    /// </summary>
    [XmlElement("code", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? Code { get; set; }
}

/// <summary>
/// Typ der Rechtsschutzversicherung.<br/>
/// <u><b>Legal protection insurance type:</b></u> Legal protection insurance type.
/// </summary>
public class RechtschutzVersicherungsTyp
{
    /// <summary>
    /// Der Code.<br/>
    /// <u><b>Code:</b></u> The code.
    /// </summary>
    [XmlElement("code", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? Code { get; set; }
}

/// <summary>
/// Fahrzeugdaten.<br/>
/// <u><b>Vehicle:</b></u> Vehicle data.
/// </summary>
public class Fahrzeug
{
    /// <summary>
    /// Das Fahrzeugkennzeichen.<br/>
    /// <u><b>License plate:</b></u> The license plate.
    /// </summary>
    [XmlElement("fahrzeugkennzeichen", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? Fahrzeugkennzeichen { get; set; }

    /// <summary>
    /// Der Typ der Fahrzeugversicherung.<br/>
    /// <u><b>Vehicle insurance type:</b></u> The vehicle insurance type.
    /// </summary>
    [XmlElement("fahrzeugversicherungstyp", Namespace = SchemeConstants.XJustizCore_Tns)]
    public FahrzeugversicherungsTyp? Fahrzeugversicherungstyp { get; set; }
}

/// <summary>
/// Typ der Fahrzeugversicherung.<br/>
/// <u><b>Vehicle insurance type:</b></u> Vehicle insurance type.
/// </summary>
public class FahrzeugversicherungsTyp
{
    /// <summary>
    /// Der Code.<br/>
    /// <u><b>Code:</b></u> The code.
    /// </summary>
    [XmlElement("code", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? Code { get; set; }
}

/// <summary>
/// Selbstbeteiligung.<br/>
/// <u><b>Deductible:</b></u> Deductible.
/// </summary>
public class Selbstbeteiligung
{
    /// <summary>
    /// Selbstbehalt.<br/>
    /// <u><b>Deductible amount:</b></u> Deductible amount.
    /// </summary>
    [XmlElement("selbstbehaltbeteiligung", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? SelbstbehaltBeteiligung { get; set; }

    /// <summary>
    /// Teilweise umfassende Selbstbeteiligung.<br/>
    /// <u><b>Partially comprehensive deductible:</b></u> Partially comprehensive deductible.
    /// </summary>
    [XmlElement("teilweiseumfassendeSelbstbeteiligung", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? TeilweiseUmfassendeSelbstbeteiligung { get; set; }

    /// <summary>
    /// Umfassende Selbstbeteiligung.<br/>
    /// <u><b>Comprehensive deductible:</b></u> Comprehensive deductible.
    /// </summary>
    [XmlElement("eumfassendeSelbstbeteiligung", Namespace = SchemeConstants.XJustizCore_Tns)]
    public string? UmfassendeSelbstbeteiligung { get; set; }
}

File: java/src/main/java/de/xjustiz/core/models/VersicherungCore.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlElement;
import jakarta.xml.bind.annotation.XmlElementWrapper;
import jakarta.xml.bind.annotation.XmlElements;
import jakarta.xml.bind.annotation.XmlType;
import java.util.List;
import java.util.UUID;
import org.jetbrains.annotations.Nullable;

/**
 * Repräsentiert eine VersicherungCore.
 * <p>
 * <u><b>Insurance:</b></u> Represents an insurance.
 */
@XmlType(name = "VersicherungCore", propOrder = { "versicherungsnummer", "versicherungsunternehmen", "anschrift", "referenzId", "versicherter", "schadensnummer", "versicherungstyp", "selbstbeteiligung", "anwendungsspezifischeErweiterung" })
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class VersicherungCore {

    /**
     * Die Versicherungsnummer.
     * <p>
     * <u><b>Insurance number:</b></u> The insurance number.
     */
    @XmlElement(name = "versicherungsnummer", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "versicherungsnummer", namespace = "http://www.xjustizcore.de")
    @JsonProperty("Versicherungsnummer")
    @JsonAlias({ "versicherungsnummer" })
    @Nullable
    private String versicherungsnummer;

    /**
     * Liste der Versicherungsunternehmen.
     * <p>
     * <u><b>Insurance companies:</b></u> List of insurance companies.
     */
    @XmlElement(name = "versicherungsunternehmen", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "versicherungsunternehmen", namespace = "http://www.xjustizcore.de")
    @JacksonXmlElementWrapper(useWrapping = false)
    @JsonProperty("Versicherungsunternehmen")
    @JsonAlias({ "versicherungsunternehmen" })
    @Nullable
    private List<Organisation> versicherungsunternehmen;

    /**
     * Die Anschrift der VersicherungCore.
     * <p>
     * <u><b>Address:</b></u> The address of the insurance.
     */
    @XmlElement(name = "anschrift", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "anschrift", namespace = "http://www.xjustizcore.de")
    @JsonProperty("Anschrift")
    @JsonAlias({ "anschrift" })
    @Nullable
    private Anschrift anschrift;

    /**
     * Die Referenz-ID.
     * <p>
     * <u><b>Reference ID:</b></u> The reference ID.
     */
    @XmlElement(name = "referenzid", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "referenzid", namespace = "http://www.xjustizcore.de")
    @JsonProperty("ReferenzId")
    @JsonAlias({ "referenzId" })
    @Nullable
    private UUID referenzId;

    /**
     * Der Versicherte.
     * <p>
     * <u><b>Insured person:</b></u> The insured person.
     */
    @XmlElement(name = "versicherter", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "versicherter", namespace = "http://www.xjustizcore.de")
    @JsonProperty("Versicherter")
    @JsonAlias({ "versicherter" })
    @Nullable
    private AuswahlBeteiligter versicherter;

    /**
     * Die Schadensnummer.
     * <p>
     * <u><b>Claim number:</b></u> The claim number.
     */
    @XmlElement(name = "schadensnummer", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "schadensnummer", namespace = "http://www.xjustizcore.de")
    @JsonProperty("Schadensnummer")
    @JsonAlias({ "schadensnummer" })
    @Nullable
    private String schadensnummer;

    /**
     * Der Typ der VersicherungCore.
     * <p>
     * <u><b>Insurance type:</b></u> The type of insurance.
     */
    @XmlElement(name = "versicherungstyp", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "versicherungstyp", namespace = "http://www.xjustizcore.de")
    @JsonProperty("Versicherungstyp")
    @JsonAlias({ "versicherungstyp" })
    @Nullable
    private VersicherungsTyp versicherungstyp;

    /**
     * Informationen zur Selbstbeteiligung.
     * <p>
     * <u><b>Deductible:</b></u> Information about the deductible.
     */
    @XmlElement(name = "selbstbeteiligung", namespace = "http://www.xjustizcore.de")
    @JacksonXmlProperty(localName = "selbstbeteiligung", namespace = "http://www.xjustizcore.de")
    @JsonProperty("Selbstbeteiligung")
    @JsonAlias({ "selbstbeteiligung" })
    @Nullable
    private Selbstbeteiligung selbstbeteiligung;

    /**
     * Anwendungsspezifische Erweiterungen.
     * <p>
     * <u><b>Application-specific extensions:</b></u> Application-specific extensions.
     */
    @XmlElement(name = "anwendungsspezifischeErweiterung", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "anwendungsspezifischeErweiterung", namespace = "http://www.xjustiz.de")
    @JsonProperty("AnwendungsspezifischeErweiterung")
    @JsonAlias({ "anwendungsspezifischeErweiterung" })
    @Nullable
    private AnwendungsspezifischeErweiterung anwendungsspezifischeErweiterung;

    public VersicherungCore() {}

    public String getVersicherungsnummer() { return versicherungsnummer; }
    public void setVersicherungsnummer(String versicherungsnummer) { this.versicherungsnummer = versicherungsnummer; }

    public List<Organisation> getVersicherungsunternehmen() { return versicherungsunternehmen; }
    public void setVersicherungsunternehmen(List<Organisation> versicherungsunternehmen) { this.versicherungsunternehmen = versicherungsunternehmen; }

    public Anschrift getAnschrift() { return anschrift; }
    public void setAnschrift(Anschrift anschrift) { this.anschrift = anschrift; }

    public UUID getReferenzId() { return referenzId; }
    public void setReferenzId(UUID referenzId) { this.referenzId = referenzId; }

    public AuswahlBeteiligter getVersicherter() { return versicherter; }
    public void setVersicherter(AuswahlBeteiligter versicherter) { this.versicherter = versicherter; }

    public String getSchadensnummer() { return schadensnummer; }
    public void setSchadensnummer(String schadensnummer) { this.schadensnummer = schadensnummer; }

    public VersicherungsTyp getVersicherungstyp() { return versicherungstyp; }
    public void setVersicherungstyp(VersicherungsTyp versicherungstyp) { this.versicherungstyp = versicherungstyp; }

    public Selbstbeteiligung getSelbstbeteiligung() { return selbstbeteiligung; }
    public void setSelbstbeteiligung(Selbstbeteiligung selbstbeteiligung) { this.selbstbeteiligung = selbstbeteiligung; }

    public AnwendungsspezifischeErweiterung getAnwendungsspezifischeErweiterung() { return anwendungsspezifischeErweiterung; }
    public void setAnwendungsspezifischeErweiterung(AnwendungsspezifischeErweiterung anwendungsspezifischeErweiterung) { this.anwendungsspezifischeErweiterung = anwendungsspezifischeErweiterung; }

}